### PR TITLE
API: Unrestricted rule for home dealers

### PR DESCRIPTION
**Pull request by flutistar**
*Issue #:* /bcgov/entity#24056
*Description of changes:*

- Removed location restrictions on transport permits
- sole owner restrictions on transfer due to sale or gift

NOTE: Since I am not comfortable to update the DB yet, I left the other 2 tasks.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: mhr-api/src/mhr_api/utils/registration_validator.py [hunk 1]
```diff
@@ -276,16 +276,17 @@ def validate_permit(
             error_msg += validator_utils.validate_registration_state(
                 registration, staff, MhrRegistrationTypes.PERMIT, MhrDocumentTypes.REG_103, json_data
             )
-            if registration and group_name and group_name == MANUFACTURER_GROUP:
```
Comment (doug-lovett) [inline mhr-api/src/mhr_api/utils/registration_validator.py:279]: From the ticket description the removal of the restriction applies only to dealers. MANUFACTURER_GROUP applies to QS manufactures, which I do not think has changed.

File: mhr-api/tests/unit/utils/test_permit_validator.py [hunk 2]
```diff
@@ -376,18 +376,18 @@
     ('Missing land status confirm MH_PARK', False, '000900', LOCATION_PARK_2,
      validator_utils.STATUS_CONFIRMATION_REQUIRED,
      REQUEST_TRANSPORT_PERMIT),
-    ('MANUFACTURER no existing lot', False, '000919', None, validator.MANUFACTURER_DEALER_INVALID,
```
Comment (doug-lovett) [inline mhr-api/tests/unit/utils/test_permit_validator.py]: If the validation change only applies to QS dealers then the manufacturer tests can remain as is.

**Review by doug-lovett — COMMENTED** [review]
Looks good, just a comment on scope - if QS manufacturer validation is changing? From the ticket description it looks like only QS dealer validation is changing.

File: mhr-api/src/mhr_api/utils/registration_validator.py (hunk 1 above)
Comment (cameron-eyds) [reply mhr-api/src/mhr_api/utils/registration_validator.py:279]: Definitely correct, only Dealers are changing.

Comment (flutistar) [reply mhr-api/src/mhr_api/utils/registration_validator.py:279]: Yep! I will update it.

File: mhr-api/tests/unit/utils/test_permit_validator.py (hunk 2 above)
Comment (flutistar) [reply mhr-api/tests/unit/utils/test_permit_validator.py]: I will revert this change!

**Pull request closed without merging**